### PR TITLE
kmymoney: 5.1.1 → 5.1.3

### LINKED ISSUE #182499: kmymoney fails to build
### Steps To Reproduce
1. `$ nix-build '<nixpkgs>' -A kmymoney`

### Build log
https://gist.github.com/aidalgol/f19c931b1f46e9f957c39a7a555f953b
(Errors on lines [6327](https://gist.github.com/aidalgol/f19c931b1f46e9f957c39a7a555f953b#file-gistfile1-txt-L6327=) and [6331](https://gist.github.com/aidalgol/f19c931b1f46e9f957c39a7a555f953b#file-gistfile1-txt-L6327=))

### Notify maintainers

No maintainers listed.  @roberth last modified this package a year ago.

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
$ nix-shell -p nix-info --run "nix-info -m"
this path will be fetched (0.06 MiB download, 0.30 MiB unpacked):
  /nix/store/qx63l5swii7wll35xfgj7jqhycf9iy19-bash-interactive-5.1-p16-dev
copying path '/nix/store/qx63l5swii7wll35xfgj7jqhycf9iy19-bash-interactive-5.1-p16-dev' from 'https://cache.nixos.org'...
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.53, NixOS, 22.05 (Quokka), 22.05.20220714.c06d5fa`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.8.1`
 - channels(aidan): `"nixpkgs"`
 - channels(root): `"nixos"`
 - nixpkgs: `/home/aidan/.nix-defexpr/channels/nixpkgs`
```

### PR DESCRIPTION
**Pull request by das-g**
fixes #182499

###### Description of changes

- bump version to 5.1.3 ([announcement of 5.1.2](https://kmymoney.org/2021/06/23/kmymoney-5-1-2-released.html), [announcement of 5.1.3](https://kmymoney.org/2022/07/30/kmymoney-5-1-3-released.html))
  - this includes [a fix](https://invent.kde.org/office/kmymoney/-/commit/265d4dc3d1dc8592ec1122aa04bfb7d431a43d5d) for #182499 (upstream [Bug 443208](https://bugs.kde.org/show_bug.cgi?id=443208))
- replace `weboob` with `woob` everywhere in `pkgs/applications/office/kmymoney/default.nix`
- patch the WOOB Python script already in the source tree before building (in `postPatch`) rather than after building (`in postInstall`) in the build result `$out`, as it's now embedded into a binary rather than visible in the build result as a separate file.
- remove Python 2 from dependencies, as it isn't needed anymore
- make `libpython` available to `.kmymoney-wrapped`, so that the Python interpreter now embedded in that binary can work

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).
  - [x] `meta.description` according to [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md)
  - [x] `meta.license` fits upstream license (confirmed by upstream [in their Telegram group](https://t.me/KMyMoney/10763))
  - [x] `meta.maintainers` set

### CODE REVIEW

**Comment by aidalgol** [comment]
I found an [answer on StackOverflow](https://stackoverflow.com/questions/67891197/ctypes-cpython-39-x86-64-linux-gnu-so-undefined-symbol-pyfloat-type-in-embedd) that sounds relevant to the `ImportError` from the woob plugin.  I am still trying to fully understand the details myself, though, so it may be a similar symptom but a different root cause.

**Comment by aidalgol** [comment]
I think we can safely drop `python2` from `nativeBuildInputs` now.

**Comment by aidalgol** [comment]
@das-g I have a fix for this PR, which I have submitted as a PR to the branch on your `nixpkgs` fork.

**Comment by das-g** [comment]
Thanks, @aidalgol, your changes seem to have solved the remaining issues:
```
Loading "/nix/store/xgfc5pdnjpm18xi3f0pcc0nbnlm101qp-kmymoney-5.1.3/lib/qt-5.15.5/plugins/kmymoney/woob.so"
Python interpreter found: 3.10.5 (main, Jun  6 2022, 12:05:50) [GCC 11.3.0]
Saved a copy of the embedded :/plugins/woob/kmymoneywoob.py script as "/run/user/1000/kmymoneyImGAhg.py"
Attempt to load the "kmymoneyImGAhg" Python module from "/run/user/1000"
The dependency 'sys' Python module loaded successfully
The 'path' Python variable loaded successfully
"kmymoneyImGAhg" Python module loaded successfully
Plugins: woob loaded
```

This package currently has no maintainers listed. Would you be willing to maintain it together with me, @aidalgol?

File: pkgs/applications/office/kmymoney/default.nix [hunk 1]
```diff
@@ -62,6 +63,13 @@ stdenv.mkDerivation rec {
         ARGS="-E '(reports-chart-test)'" # Test fails, so exclude it for now.
     '';
 
+  # libpython is required by the python interpreter embedded in kmymoney, so we
+  # need to explicitly tell autoPatchelf about it.
+  postFixup = ''
+    patchelf --debug --add-needed libpython3.10.so \
```
Comment (SuperSandro2000) [inline pkgs/applications/office/kmymoney/default.nix:69]:
```suggestion
    patchelf --debug --add-needed libpython${python3.pythonVersion}.so \
```

File: pkgs/applications/office/kmymoney/default.nix [hunk 2]
```diff
@@ -38,20 +39,20 @@ stdenv.mkDerivation rec {
 
     # Put it into buildInputs so that CMake can find it, even though we patch
     # it into the interface later.
-    python3Packages.weboob
+    python3Packages.woob
   ];
 
-  weboobPythonPath = [ python3Packages.weboob ];
+  woobPythonPath = [ python3Packages.woob ];
```
[the remark this answers is not in the document]
Comment (das-g) [reply pkgs/applications/office/kmymoney/default.nix:46]: Will `buildPythonPath "$woobPythonPath"` [in `postPatch`](https://github.com/NixOS/nixpkgs/pull/183973/files#diff-4004fbc3fdb7ac27f994a39aaf8785f18889895120945e21fb984e7e338afc59R48) keep working without that?

File: pkgs/applications/office/kmymoney/default.nix [hunk 3]
```diff
@@ -10,24 +11,24 @@
 # Needed for running tests:
 , qtbase, xvfb-run
```
Comment (das-g) [inline pkgs/applications/office/kmymoney/default.nix:13]:
```suggestion

, python3
```
(needed for https://github.com/NixOS/nixpkgs/pull/183973/files#r935577346)

File: pkgs/applications/office/kmymoney/default.nix [hunk 4]
```diff
@@ -10,24 +11,25 @@
...
 
-, python2, python3Packages
+, python3
+, python3Packages
```
Comment (SuperSandro2000) [inline pkgs/applications/office/kmymoney/default.nix:15]:
```suggestion
```

File: pkgs/applications/office/kmymoney/default.nix [hunk 5]
```diff
@@ -38,20 +40,20 @@ stdenv.mkDerivation rec {
 
     # Put it into buildInputs so that CMake can find it, even though we patch
     # it into the interface later.
-    python3Packages.weboob
+    python3Packages.woob
```
Comment (SuperSandro2000) [inline pkgs/applications/office/kmymoney/default.nix:43]:
```suggestion
    python3.pkgs.woob
```

File: pkgs/applications/office/kmymoney/default.nix [hunk 6]
```diff
@@ -38,20 +40,20 @@ stdenv.mkDerivation rec {
...
   ];
 
-  weboobPythonPath = [ python3Packages.weboob ];
+  woobPythonPath = [ python3Packages.woob ];
 
-  postInstall = ''
-    buildPythonPath "$weboobPythonPath"
-    patchPythonScript "$out/share/kmymoney/weboob/kmymoneyweboob.py"
+  postPatch = ''
+    buildPythonPath "$woobPythonPath"
```
Comment (SuperSandro2000) [inline pkgs/applications/office/kmymoney/default.nix:49]:
```suggestion
    buildPythonPath "${python3.pkgs.woob}"
```

File: pkgs/applications/office/kmymoney/default.nix (hunk 2 above)
[the remark this answers is not in the document]
Comment (SuperSandro2000) [reply pkgs/applications/office/kmymoney/default.nix:47]: we are inling that in there

**Comment by aidalgol** [comment]
> This package currently has no maintainers listed. Would you be willing to maintain it together with me, @aidalgol?

Certainly.

File: pkgs/applications/office/kmymoney/default.nix (hunk 2 above)
[the remark this answers is not in the document]
Comment (das-g) [reply pkgs/applications/office/kmymoney/default.nix]: applied with b371f5b5416643a42bdcd7fe8cf97dd07c191206

File: pkgs/applications/office/kmymoney/default.nix (hunk 6 above)
Comment (das-g) [reply pkgs/applications/office/kmymoney/default.nix]: applied with b371f5b5416643a42bdcd7fe8cf97dd07c191206 & 1234550f536a6f72069004b1cceb80fde3d8ffdc

File: pkgs/applications/office/kmymoney/default.nix (hunk 4 above)
Comment (das-g) [reply pkgs/applications/office/kmymoney/default.nix]: applied with 1234550f536a6f72069004b1cceb80fde3d8ffdc

File: pkgs/applications/office/kmymoney/default.nix (hunk 5 above)
Comment (das-g) [reply pkgs/applications/office/kmymoney/default.nix]: applied with 1234550f536a6f72069004b1cceb80fde3d8ffdc

**Review by aidalgol — APPROVED** [review]

**Comment by das-g** [comment]
Result of `nixpkgs-review pr 183973` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>kmymoney</li>
  </ul>
</details>

**Pull request merged**